### PR TITLE
Init authentication server

### PR DESCRIPTION
**Pull request by mneira10**
Hey! I managed to figure out how to do authentication. We still need to decide if we want to do JWT/session cookies but we've figured out how to authenticate someone with a public/private key pair off chain! 🎉

My rust is very primitive - all feedback is super welcome! Let me know what you think!

Example output from GET `localhost:8080/signature_message/my_pub_key`:

```
strangemood.org wants you to sign in with your Ethereum account:
my_pub_key

I accept the Strangemood Terms of Service.

URI: strangemood.org/login
Version: 0.0.0
Nonce: snhho56cgu08pgap5zgqp9tym4jr9dwoembpskw8vmlomqxwel3fm2q983tohhmw
Issued At: 2021-12-10 08:26:30
```

Still some work to be done but its a good first step!

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -1,5 +1,6 @@
 [workspace]
+exclude = ["auth/server"]
```
Comment (mneira10) [inline Cargo.toml:2]: I'm a crate newbie but this seems like a bad thing

File: auth/README.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+2. User signs the message string with their private key and sends the message AND the signature as a POST request to `/login`
+3. Server recieves the request, verifies the signature _and the nonce_
+4. If Signature and nonce are valid, the user is given a session or JWT
+
+Since the user will see the message as plain text, its important that it is not in machine readable code like `json` or `xml`. We will need to parse the data to retrieve the relevant fields.
+
+This is the strangemood message format:
+
+```
+${domain} wants you to sign in with your Ethereum account:
```
Comment (Flaque) [inline auth/README.md:15]: Solana* :P

Comment (mneira10) [reply auth/README.md:15]: LOL

**Review by Flaque — APPROVED** [review]
This lgtm to me btw

**Pull request merged**